File: TransCoder_tokenized_test_set_functions/cpp/K_TH_DISTINCT_OR_NON_REPEATING_ELEMENT_IN_AN_ARRAY_1.cpp
int printKDistinct ( int arr [ ], int n, int k ) {
  unordered_map < int, int > h;
  for ( int i = 0;
  i < n;
  i ++ ) h [ arr [ i ] ] ++;
  if ( h . size ( ) < k ) return - 1;
  int dist_count = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( h [ arr [ i ] ] == 1 ) dist_count ++;
    if ( dist_count == k ) return arr [ i ];
  }
  return - 1;
}

File: results/java_codex/K_TH_DISTINCT_OR_NON_REPEATING_ELEMENT_IN_AN_ARRAY_1.java
int printKDistinct(int arr[], int n, int k) {
  unordered_map<int, int> h;
  for (int i = 0; i < n; i++) {
    if (h.find(arr[i]) != h.end()) h[arr[i]]++;
    else h[arr[i]] = 1;
  }
  if (h.size() < k) return -1;
  int dist_count = 0;
  for (int i = 0; i < n; i++) {
    if (h[arr[i]] == 1) dist_count++;
    if (dist_count == k) return arr[i];
  }
  return -1;
}